x-pack/plugins/actions/server/builtin_action_types/resilient/service.test.ts

```diff
@@ -0,0 +1,422 @@
+    });
+  } else {
+    requestMock.mockImplementationOnce(() => ({
+      data: {
+        success: true,
+        id: '1',
+        inc_last_modified_date: 1589391874472,
+      },
+    }));
+  }
+
+  requestMock.mockImplementationOnce(() => ({
+    data: {
+      id: '1',
+      name: 'title_updated',
+      description: {
+        format: 'html',
+        content: 'desc_updated',
+      },
+      inc_last_modified_date: 1589391874472,
+    },
+  }));
+};
+
+describe('IBM Resilient service', () => {
+  let service: ExternalService;
+
+  beforeAll(() => {
+    service = createExternalService({
+      config: { apiUrl: 'https://resilient.elastic.co', orgId: '201' },
+      secrets: { apiKeyId: 'keyId', apiKeySecret: 'secret' },
+    });
+  });
+
+  afterAll(() => {
+    Date.now = now;
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    Date.now = jest.fn().mockReturnValue(TIMESTAMP);
+  });
+
+  describe('getValueTextContent', () => {
+    test('transforms correctly', () => {
+      expect(getValueTextContent('name', 'title')).toEqual({
+        text: 'title',
+      });
+    });
+
+    test('transforms correctly the description', () => {
+      expect(getValueTextContent('description', 'desc')).toEqual({
+        textarea: {
+          format: 'html',
+          content: 'desc',
+        },
+      });
+    });
+  });
+
+  describe('formatUpdateRequest', () => {
+    test('transforms correctly', () => {
+      const oldIncident = { name: 'title', description: 'desc' };
+      const newIncident = { name: 'title_updated', description: 'desc_updated' };
+      expect(formatUpdateRequest({ oldIncident, newIncident })).toEqual({
+        changes: [
+          {
+            field: { name: 'name' },
+            old_value: { text: 'title' },
+            new_value: { text: 'title_updated' },
+          },
+          {
+            field: { name: 'description' },
+            old_value: {
+              textarea: {
+                format: 'html',
+                content: 'desc',
+              },
+            },
+            new_value: {
+              textarea: {
+                format: 'html',
+                content: 'desc_updated',
+              },
+            },
+          },
+        ],
+      });
+    });
+  });
+
+  describe('createExternalService', () => {
+    test('throws without url', () => {
+      expect(() =>
+        createExternalService({
+          config: { apiUrl: null, orgId: '201' },
+          secrets: { apiKeyId: 'token', apiKeySecret: 'secret' },
+        })
+      ).toThrow();
+    });
+
+    test('throws without orgId', () => {
+      expect(() =>
+        createExternalService({
+          config: { apiUrl: 'test.com', orgId: null },
+          secrets: { apiKeyId: 'token', apiKeySecret: 'secret' },
+        })
+      ).toThrow();
+    });
+
+    test('throws without username', () => {
+      expect(() =>
+        createExternalService({
+          config: { apiUrl: 'test.com' },
```

Comment:
couldn't this be throwing because no orgId? i think you should define everything except what you are testing to throw, or it could be throwing for a different reason